### PR TITLE
typing updates for Task

### PR DESCRIPTION
**Pull request by albertz**
Also removed Python 2 style code to derive from object.

I did not cover `update_rqmt` yet because this would require `Protocol` (not just `Callable`), and I'm not sure if we want that (requires Python 3.8).

### CODE REVIEW

File: sisyphus/task.py [hunk 1]
```diff
@@ -3,35 +3,35 @@
...
         self,
-        start,
-        resume=None,
-        rqmt=None,
-        args=None,
-        mini_task=False,
+        start: str,
+        resume: Optional[str] = None,
+        rqmt: Optional[Dict[str, Any]] = None,
+        args: Optional[Sequence[Union[List[Any], Any]]] = None,
```
Comment (michelwi) [inline sisyphus/task.py:24]:
```suggestion
        args: Optional[Sequence[Any]] = None,
```
can already be `Any` and `List[Any]` is contained in `Any`.

**Comment by michelwi** [comment]
>  did not cover update_rqmt yet because this would require Protocol (not just Callable), and I'm not sure if we want that

Why would `Callable[[Dict[str, Any], Dict[str, Any]], Dict[str, Any]]` not work?

File: sisyphus/task.py (hunk 1 above)
Comment (albertz) [reply sisyphus/task.py:24]: Yea I know, but sometimes I prefer this to give some more specific type hints to what types would be specifically expected here, or specially handled, even though other types (any other types) would also be ok.

Also, I basically just translated the typing from the earlier docstring. I also wondered a bit about why this is here at all, and I thought there is indeed maybe a special meaning when there is a list?

**Comment by albertz** [comment]
> > did not cover update_rqmt yet because this would require Protocol (not just Callable), and I'm not sure if we want that
> 
> Why would `Callable[[Dict[str, Any], Dict[str, Any]], Dict[str, Any]]` not work?

We call `self._update_rqmt(last_rqmt=last_rqmt, last_usage=last_usage)`, i.e. we use named args, which is not supported by `Callable`.

**Review by michelwi — APPROVED** [review]

**Review by NeoLegends — APPROVED** [review]
👍

**Pull request merged**